### ISSUE TITLE
The entropy weights function in code is not equal to it in paper

### ISSUE THREAD

**Issue by jjore95** [zhangyan-ucas/GUI-SD-code#3]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

opsd_trainer.py line 263
               vocab_size = outputs_teacher.logits.size(-1)
                H_max = torch.log(torch.tensor(vocab_size, dtype=token_entropy.dtype, device=token_entropy.device))
                confidence = 1.0 - token_entropy / H_max  # [N_valid], 范围 [0, 1
entropy weights in paper is a gating function as is shown in the image

<img width="726" height="214" alt="Image" src="https://github.com/user-attachments/assets/e27aaa45-0a48-48f0-87b3-85f86ddeecac" />

**Comment by zhangyan-ucas** [comment]
Thanks for catching this! This inconsistency was introduced while preparing the code for release. Our original implementation follows Eq. (6) in the paper, `exp(-H/τ)`. We will correct the released code accordingly. Sorry for the confusion!
